sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_models.py

```diff
@@ -53,3 +55,39 @@ class SettingSelector:
     def __init__(self, *, key_filter: str, label_filter: Optional[str] = EMPTY_LABEL):
         self.key_filter = key_filter
         self.label_filter = label_filter
+
+
+class RefreshRegistrations:
+    def __init__(
+        self, key_filter: str, label_filter: Optional[str], refresh_all: Optional[bool]
+    ):
+        self.key_filter = key_filter
+        self.label_filter = label_filter
+        self.refresh_all = refresh_all
+        self.etag = None
+
+
+class AzureAppConfigurationRefreshOptions:
```

Comment:
Yep this whole object looks copy+pasted from .NET. Let's get rid of it.

Reply:
This might be moot by the above comment, but this was already the naming convention used with the Key Vault options.

Reply:
Sorry I should have clarified - my comment was not so much about the name of the object, but the structure and nature of using an object for this at all - especially the "setter" method, which is not really a design pattern used in Python.

(While I still regret the name of the Keyvault options object, and would like the deprecate this is future if I can find a nice way to do it, using "Azure" as a prefix made a bit more sense there as it was a reference to a separate service external to the current SDK).